x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_activity_flyout/flyout_body.tsx

```diff
@@ -57,6 +63,13 @@ export const FlyoutBody: React.FunctionComponent<{
   onChangeDateFilter,
   agentPolicies,
 }) => {
+  const scrollToTopRef = React.useRef<HTMLDivElement>(null);
+  React.useEffect(() => {
+    // Condition needed for jest tests as scrollIntoView is not implemented in jsdom
+    if (scrollToTopRef.current?.scrollIntoView) {
+      scrollToTopRef.current.scrollIntoView();
+    }
+  }, [dateFilter]);
```

Comment:
is there any chance of this object mutating due to re-rendering, but without the actual date being changed?

Reply:
@jen-huang I think it's safe `dateFilter` is stored in a react state and only mutated when there is an actual date change